pkg/tt/mux/mux.go

```diff
@@ -0,0 +1,66 @@
+package mux
+
+import (
+	"context"
+	"sync"
+
+	"github.com/kris-nova/T_T/pkg/tt"
+)
+
+// Mux is a router for action-based plugins
+type Mux interface {
+	tt.Handler
+
+	Register(p tt.Plugin)
+}
+
+// New creates a new Mux router
+func New() Mux {
+	return &mux{
+		plugins: make(map[int64]tt.Plugin),
+	}
+}
+
+type mux struct {
+	l       sync.RWMutex
+	rules   []tt.Rule
+	plugins map[int64]tt.Plugin
+}
+
+func (mx *mux) Register(p tt.Plugin) {
+	rl := p.Rules()
+
+	mx.l.Lock()
```

Comment:
Yes. RWMutex. Full Read-Write lock during registration of plugins (usually on system startup) and Read lock during message routing. Shouldn't be an issue.